Fullstacker/Api/Controllers/InventoryController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using API.Dtos;
+using DataLayer.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers
+{
+    [Route("api/[controller]")]
+    public class InventoryController : ControllerBase
+    {
+        private readonly IInventoryRepository _inventoryRepository;
+
+        public InventoryController(IInventoryRepository inventoryRepository)
+        {
+            _inventoryRepository = inventoryRepository;
+        }
+        /// <summary>
+        /// An endpoint for retrieving a list of inventories
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("get")]
+        public async Task<IActionResult> Get()
+        {
+            var inventory = await _inventoryRepository.Get();
+
+            if (inventory is null || !inventory.Any())
+                return NotFound("No items found.");
+
+            return Ok(inventory);
+        }
+
+        /// <summary>
+        /// An endpoint for retrieving an inventory by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("get/{id}")]
+        public async Task<IActionResult> Get(int id)
+        {
+            var inventory = await _inventoryRepository.Get(id);
+            if (inventory is null)
+                return NotFound($"No inventory found with id '{id}'");
+
+            return Ok(inventory);
+        }
+
+        /// <summary>
+        /// An endpoint for creating an inventory
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("create")]
+        public async Task<IActionResult> Create([FromBody] CreateInventoryDto request)
+        {
+            if (string.IsNullOrWhiteSpace(request.ItemId.ToString()) || string.IsNullOrWhiteSpace(request.Quantity.ToString()))
```

Comment:
don't tostring an int. If it's an int. Besides, it can never be null. Research value types vs reference types.  Negative int will be <= 0

Reply:
Same with quantity.